File: event-service/src/main/java/com/example/event_service/model/Event.java
package com.example.event_service.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "events")
public class Event {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String title;
    private String description;
    private String location;
    private LocalDate date;
    private String time;
    private TYPE type;
}

File: frontend/src/app/model/event.ts
export interface Event {
  id?: number;
  title: string;
  description: string;
  location: string;
  date: string;
  time: string;
  type: EventType;
  owner: string;
}

export enum EventType {
  ACADEMIC = 'ACADEMIC',
  SPORT = 'SPORT',
  CULTURAL = 'CULTURAL'
}
